server/src/actors/events.rs

```diff
@@ -456,24 +453,32 @@ impl Handler<HandleEvent> for EventManager {
             start_time,
         } = message;
 
+        let project_id_for_err = Arc::new(Mutex::new(None::<u64>));
+        let org_id_for_err = Arc::new(Mutex::new(None::<u64>));
+
         let future = project
             .send(GetProjectId)
             .map(One::into_inner)
             .map_err(ProcessingError::ScheduleFailed)
-            .and_then(move |project_id| {
+            .and_then(clone! { project_id_for_err, org_id_for_err, |project_id| {
+                *project_id_for_err.lock() = Some(project_id);
                 project
                     .send(GetEventAction::fetched(meta.clone()))
                     .map_err(ProcessingError::ScheduleFailed)
                     .and_then(|action| match action.map_err(ProcessingError::NoAction)? {
                         EventAction::Accept => Ok(()),
-                        EventAction::RetryAfter(s) => Err(ProcessingError::RateLimited(s)),
-                        EventAction::Discard => Err(ProcessingError::EventRejected),
+                        EventAction::RetryAfter(secs, reason) => {
+                            Err(ProcessingError::RateLimited(secs, reason))
+                        }
+                        EventAction::Discard(reason) => Err(ProcessingError::EventRejected(reason)),
                     })
                     .and_then(clone!(project, |_| project
                         .send(GetProjectState)
                         .map_err(ProcessingError::ScheduleFailed)
                         .and_then(|result| result.map_err(ProcessingError::PiiFailed))))
-                    .and_then(clone!(meta, |project_state| processor
+                    .and_then(clone! (meta,  |project_state| {
```

Comment:
_nit_: Seems like there are too many spaces here that are not formatted by `rustfmt` due to the macro.

Reply:
ok, done